src/libraries/Microsoft.Extensions.Http/src/ExpiredHandlerTrackingEntry.cs

```diff
@@ -23,12 +25,33 @@ public ExpiredHandlerTrackingEntry(ActiveHandlerTrackingEntry other)
             InnerHandler = other.Handler.InnerHandler!;
         }
 
+        // Used during normal cleanup cycles
         public bool CanDispose => !_livenessTracker.IsAlive;
 
         public HttpMessageHandler InnerHandler { get; }
 
         public string Name { get; }
 
         public IServiceScope? Scope { get; }
+
+        public void Dispose()
+        {
+            // Always dispose the inner handler when explicitly disposing or when CanDispose is true
+            if (_forceDispose || !_livenessTracker.IsAlive)
+            {
+                try
+                {
+                    InnerHandler.Dispose();
+                }
+                finally
+                {
+                    Scope?.Dispose();
+                }
+            }
+            else
+            {
+                Scope?.Dispose();
```

Comment:
no, we shouldn't dispose scope here; if IsAlive=true, it means it's being used together with the handler.